`test_no_stale_references` fails on `collect.wait` timeout on CPython 3.14.0rc2t

```python
$ pytest -vlx -o faulthandler_timeout=30 -k "test_no_stale_references and Loky"
================================================================================= test session starts =================================================================================
platform darwin -- Python 3.14.0rc2, pytest-8.4.1, pluggy-1.6.0 -- /Users/ogrisel/.venvs/py3140rc2t/bin/python3
cachedir: .pytest_cache
rootdir: /Users/ogrisel/code/loky
configfile: pyproject.toml
collected 319 items / 318 deselected / 1 selected                                                                                                                                     

tests/test_process_executor_loky.py::TestsProcessPoolLokyExecutor::test_no_stale_references FAILED                                                                              [100%]

====================================================================================== FAILURES =======================================================================================
________________________________________________________________ TestsProcessPoolLokyExecutor.test_no_stale_references ________________________________________________________________

self = <tests.test_process_executor_loky.TestsProcessPoolLokyExecutor object at 0x4555e5ad210>

    @pytest.mark.skipif(
        platform.python_implementation() != "CPython",
        reason="cpython specific non-regression test (fixed starting 3.8.2)",
    )
    def test_no_stale_references(self):
        # Issue #16284: check that the executors don't unnecessarily hang onto
        # references.
    
        # This test has to be skipped on early Python 3.8 versions because of a
        # low-level reference cycle inside the pickle module for early versions
        # of Python 3.8 preventing stale references from being collected. See
        # cloudpipe/cloudpickle#327 as well as
        # https://bugs.python.org/issue39492
        my_object = MyObject()
        collect = threading.Event()
        _ref = weakref.ref(my_object, lambda obj: collect.set())  # noqa
        # Deliberately discarding the future.
        self.executor.submit(my_object.my_method)
        del my_object
    
        collected = False
        for _ in range(5):
            collected = collect.wait(timeout=1.0)
            if collected:
                return
>       assert collected, "Stale reference not collected within timeout."
               ^^^^^^^^^
E       AssertionError: Stale reference not collected within timeout.

_          = 4
_ref       = <weakref at 0x455614cba00; to 'tests._test_process_executor.MyObject' at 0x45560e69f10>
collect    = <threading.Event at 0x455619d7f90: unset>
collected  = False
self       = <tests.test_process_executor_loky.TestsProcessPoolLokyExecutor object at 0x4555e5ad210>

tests/_test_process_executor.py:710: AssertionError
-------------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------------
setup class with <loky.backend.context.LokyContext object at 0x45560e61510>
-------------------------------------------------------------------------------- Captured stderr setup --------------------------------------------------------------------------------
[DEBUG:MainProcess:MainThread] created semlock with handle 11 and name "None"
[DEBUG:MainProcess:MainThread] launching resource tracker: ['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-c', 'from loky.backend.resource_tracker import main; main(12, False)']
[DEBUG:MainProcess:MainThread] created semlock with handle 12 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 14 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 15 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 16 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 17 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 22 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 23 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 24 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 27 and name "None"
[DEBUG:MainProcess:MainThread] created semlock with handle 28 and name "None"
[DEBUG:MainProcess:MainThread] ProcessPoolExecutor is setup
[DEBUG:MainProcess:MainThread] created semlock with handle 29 and name "None"
[DEBUG:MainProcess:MainThread] launched python with pid 16809 and cmd:
['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-1', '--pipe', '33']
[DEBUG:MainProcess:MainThread] created semlock with handle 32 and name "None"
[DEBUG:MainProcess:MainThread] launched python with pid 16810 and cmd:
['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-2', '--pipe', '35']
[DEBUG:MainProcess:MainThread] created semlock with handle 34 and name "None"
[DEBUG:MainProcess:MainThread] launched python with pid 16811 and cmd:
['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-3', '--pipe', '37']
[DEBUG:MainProcess:MainThread] created semlock with handle 36 and name "None"
[DEBUG:MainProcess:MainThread] launched python with pid 16812 and cmd:
['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-4', '--pipe', '39']
[DEBUG:MainProcess:MainThread] created semlock with handle 38 and name "None"
[DEBUG:MainProcess:MainThread] launched python with pid 16813 and cmd:
['/Users/ogrisel/.venvs/py3140rc2t/bin/python3', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-5', '--pipe', '41']
[DEBUG:MainProcess:MainThread] Adjusted process count to 5: [('LokyProcess-1', 16809), ('LokyProcess-2', 16810), ('LokyProcess-3', 16811), ('LokyProcess-4', 16812), ('LokyProcess-5', 16813)]
[DEBUG:MainProcess:MainThread] _start_executor_manager_thread called
[DEBUG:MainProcess:ExecutorManagerThread] Queue._start_thread()
[DEBUG:MainProcess:ExecutorManagerThread] doing self._thread.start()
[DEBUG:MainProcess:ExecutorManagerThread] ... done self._thread.start()
[DEBUG:MainProcess:QueueFeederThread] starting thread to feed data to pipe
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 22 and name "/loky-16806-etbxraxq"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 23 and name "/loky-16806-pahzfgmy"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 24 and name "/loky-16806-r2ce1hk9"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 27 and name "/loky-16806-lhes14ra"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 28 and name "/loky-16806-25ww2fex"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 22 and name "/loky-16806-etbxraxq"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 23 and name "/loky-16806-pahzfgmy"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 24 and name "/loky-16806-r2ce1hk9"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 27 and name "/loky-16806-lhes14ra"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 28 and name "/loky-16806-25ww2fex"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 22 and name "/loky-16806-etbxraxq"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 23 and name "/loky-16806-pahzfgmy"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 24 and name "/loky-16806-r2ce1hk9"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 27 and name "/loky-16806-lhes14ra"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 28 and name "/loky-16806-25ww2fex"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 22 and name "/loky-16806-etbxraxq"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 23 and name "/loky-16806-pahzfgmy"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 24 and name "/loky-16806-r2ce1hk9"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 27 and name "/loky-16806-lhes14ra"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 28 and name "/loky-16806-25ww2fex"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 22 and name "/loky-16806-etbxraxq"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 23 and name "/loky-16806-pahzfgmy"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 24 and name "/loky-16806-r2ce1hk9"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 27 and name "/loky-16806-lhes14ra"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 28 and name "/loky-16806-25ww2fex"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 11 and name "/loky-16806-ug6h2e5q"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 12 and name "/loky-16806-l_whchzh"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 14 and name "/loky-16806-3mtkcwp2"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 15 and name "/loky-16806-25erjebz"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 16 and name "/loky-16806-4_cjc18n"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 17 and name "/loky-16806-pwpytw0s"
[DEBUG:LokyProcess-1:MainThread] recreated blocker with handle 29 and name "/loky-16806-mjqsonf_"
[INFO:LokyProcess-1:MainThread] child process calling self.run()
[DEBUG:LokyProcess-1:MainThread] Worker started with timeout=None
[DEBUG:LokyProcess-1:MainThread] Enabling faulthandler to report tracebacks on worker crashes.
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 11 and name "/loky-16806-ug6h2e5q"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 12 and name "/loky-16806-l_whchzh"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 14 and name "/loky-16806-3mtkcwp2"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 15 and name "/loky-16806-25erjebz"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 16 and name "/loky-16806-4_cjc18n"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 11 and name "/loky-16806-ug6h2e5q"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 17 and name "/loky-16806-pwpytw0s"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 12 and name "/loky-16806-l_whchzh"
[DEBUG:LokyProcess-5:MainThread] recreated blocker with handle 38 and name "/loky-16806-k7gsm3ak"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 14 and name "/loky-16806-3mtkcwp2"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 15 and name "/loky-16806-25erjebz"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 16 and name "/loky-16806-4_cjc18n"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 17 and name "/loky-16806-pwpytw0s"
[DEBUG:LokyProcess-4:MainThread] recreated blocker with handle 36 and name "/loky-16806-6kleftvx"
[INFO:LokyProcess-4:MainThread] child process calling self.run()
[INFO:LokyProcess-5:MainThread] child process calling self.run()
[DEBUG:LokyProcess-4:MainThread] Worker started with timeout=None
[DEBUG:LokyProcess-4:MainThread] Enabling faulthandler to report tracebacks on worker crashes.
[DEBUG:LokyProcess-5:MainThread] Worker started with timeout=None
[DEBUG:LokyProcess-1:MainThread] psutil return memory size: 39731200
[DEBUG:LokyProcess-5:MainThread] Enabling faulthandler to report tracebacks on worker crashes.
-------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 11 and name "/loky-16806-ug6h2e5q"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 12 and name "/loky-16806-l_whchzh"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 14 and name "/loky-16806-3mtkcwp2"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 15 and name "/loky-16806-25erjebz"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 16 and name "/loky-16806-4_cjc18n"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 17 and name "/loky-16806-pwpytw0s"
[DEBUG:LokyProcess-3:MainThread] recreated blocker with handle 34 and name "/loky-16806-tg4fpaod"
[INFO:LokyProcess-3:MainThread] child process calling self.run()
[DEBUG:LokyProcess-3:MainThread] Worker started with timeout=None
[DEBUG:LokyProcess-3:MainThread] Enabling faulthandler to report tracebacks on worker crashes.
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 11 and name "/loky-16806-ug6h2e5q"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 12 and name "/loky-16806-l_whchzh"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 14 and name "/loky-16806-3mtkcwp2"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 15 and name "/loky-16806-25erjebz"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 16 and name "/loky-16806-4_cjc18n"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 17 and name "/loky-16806-pwpytw0s"
[DEBUG:LokyProcess-2:MainThread] recreated blocker with handle 32 and name "/loky-16806-95h34ema"
[INFO:LokyProcess-2:MainThread] child process calling self.run()
[DEBUG:LokyProcess-2:MainThread] Worker started with timeout=None
[DEBUG:LokyProcess-2:MainThread] Enabling faulthandler to report tracebacks on worker crashes.
[DEBUG:LokyProcess-4:MainThread] psutil return memory size: 41369600
------------------------------------------------------------------------------ Captured stdout teardown -------------------------------------------------------------------------------
teardown class with <loky.backend.context.LokyContext object at 0x45560e61510>
------------------------------------------------------------------------------ Captured stderr teardown -------------------------------------------------------------------------------
[DEBUG:MainProcess:MainThread] shutting down executor <loky.process_executor.ProcessPoolExecutor object at 0x45560e68a10>
[DEBUG:MainProcess:ExecutorManagerThread] releasing worker exit lock on LokyProcess-1
[DEBUG:MainProcess:ExecutorManagerThread] releasing worker exit lock on LokyProcess-2
[DEBUG:MainProcess:ExecutorManagerThread] releasing worker exit lock on LokyProcess-3
[DEBUG:MainProcess:ExecutorManagerThread] releasing worker exit lock on LokyProcess-4
[DEBUG:MainProcess:ExecutorManagerThread] releasing worker exit lock on LokyProcess-5
[DEBUG:MainProcess:ExecutorManagerThread] found 5 processes to stop
[DEBUG:MainProcess:ExecutorManagerThread] sent 5 sentinels to the call queue
[DEBUG:MainProcess:ExecutorManagerThread] closing call_queue
[DEBUG:MainProcess:ExecutorManagerThread] telling queue thread to quit
[DEBUG:MainProcess:ExecutorManagerThread] Queue.join_thread()
[DEBUG:MainProcess:ExecutorManagerThread] closing result_queue
[DEBUG:MainProcess:ExecutorManagerThread] closing thread_wakeup
[DEBUG:MainProcess:ExecutorManagerThread] joining 5 processes
[DEBUG:MainProcess:ExecutorManagerThread] joining process LokyProcess-5 with pid 16813
[DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
[INFO:LokyProcess-1:MainThread] Shutting down worker on sentinel
[INFO:LokyProcess-5:MainThread] Shutting down worker on sentinel
[DEBUG:LokyProcess-1:MainThread] Exited cleanly
[INFO:LokyProcess-3:MainThread] Shutting down worker on sentinel
[INFO:LokyProcess-1:MainThread] process exiting with exitcode 0
[INFO:LokyProcess-1:MainThread] process shutting down
[DEBUG:LokyProcess-1:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:LokyProcess-5:MainThread] Exited cleanly
[DEBUG:LokyProcess-1:MainThread] running the remaining "atexit" finalizers
[INFO:LokyProcess-5:MainThread] process exiting with exitcode 0
[INFO:LokyProcess-2:MainThread] Shutting down worker on sentinel
[DEBUG:LokyProcess-3:MainThread] Exited cleanly
[INFO:LokyProcess-5:MainThread] process shutting down
[DEBUG:LokyProcess-5:MainThread] running all "atexit" finalizers with priority >= 0
[INFO:LokyProcess-3:MainThread] process exiting with exitcode 0
[DEBUG:LokyProcess-5:MainThread] running the remaining "atexit" finalizers
[INFO:LokyProcess-3:MainThread] process shutting down
[DEBUG:LokyProcess-3:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:LokyProcess-3:MainThread] running the remaining "atexit" finalizers
[DEBUG:LokyProcess-2:MainThread] Exited cleanly
[INFO:LokyProcess-4:MainThread] Shutting down worker on sentinel
[INFO:LokyProcess-2:MainThread] process exiting with exitcode 0
[INFO:LokyProcess-2:MainThread] process shutting down
[DEBUG:LokyProcess-2:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:LokyProcess-2:MainThread] running the remaining "atexit" finalizers
[DEBUG:LokyProcess-4:MainThread] Exited cleanly
[INFO:LokyProcess-4:MainThread] process exiting with exitcode 0
[INFO:LokyProcess-4:MainThread] process shutting down
[DEBUG:LokyProcess-4:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:LokyProcess-4:MainThread] running the remaining "atexit" finalizers
[DEBUG:MainProcess:ExecutorManagerThread] joining process LokyProcess-4 with pid 16812
[DEBUG:MainProcess:ExecutorManagerThread] joining process LokyProcess-3 with pid 16811
[DEBUG:MainProcess:ExecutorManagerThread] joining process LokyProcess-2 with pid 16810
[DEBUG:MainProcess:ExecutorManagerThread] joining process LokyProcess-1 with pid 16809
[DEBUG:MainProcess:ExecutorManagerThread] executor management thread clean shutdown of 5 workers
================================================================================== warnings summary ===================================================================================
../../.venvs/py3140rc2t/lib/python3.14t/site-packages/_pytest/config/__init__.py:1474
  /Users/ogrisel/.venvs/py3140rc2t/lib/python3.14t/site-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: faulthandler_exit_on_timeout
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================== short test summary info ===============================================================================
FAILED tests/test_process_executor_loky.py::TestsProcessPoolLokyExecutor::test_no_stale_references - AssertionError: Stale reference not collected within timeout.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================================== 1 failed, 318 deselected, 1 warning in 5.38s =====================================================================
[INFO:MainProcess:MainThread] process shutting down
[DEBUG:MainProcess:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:MainProcess:MainThread] running the remaining "atexit" finalizers
```

This seems to be free-threading related because the same tests pass on 3.14.0rc2 version CPython both installed with pyenv on macOS.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -61,10 +61,16 @@ jobs:
             os: macos-latest
             PYTHON_VERSION: "3.9"
 
+          - name: linux-py314-free-threading
+            os: ubuntu-latest
+            PYTHON_VERSION: "3.14.0rc2"
+            CONDA_CHANNEL: "conda-forge/label/python_rc"
+            FREE_THREADING: "true"
           - name: linux-py313
             os: ubuntu-latest
             PYTHON_VERSION: "3.13"
             LOKY_TEST_NO_LSCPU: "true"
+            EXTRA_PACKAGES: "viztracer"
           - name: linux-py39-joblib-tests
             os: ubuntu-latest
             PYTHON_VERSION: "3.9"

--- a/continuous_integration/install.sh
+++ b/continuous_integration/install.sh
@@ -6,8 +6,20 @@ set -xe
 
 # Create new conda env
 conda config --set solver libmamba
-to_install="python=$PYTHON_VERSION pip numpy tblib viztracer"
-conda create -n testenv --yes -c conda-forge $to_install
+
+if [[ "$FREE_THREADING" == "true" ]]; then
+    PYTHON_PACKAGE="python-freethreading"
+else
+    PYTHON_PACKAGE="python"
+fi
+
+# If the conda channel is not explicitly set, use conda-forge:
+if [[ -z "$CONDA_CHANNEL" ]]; then
+    CONDA_CHANNEL="conda-forge"
+fi
+
+to_install="$PYTHON_PACKAGE=$PYTHON_VERSION pip numpy tblib $EXTRA_PACKAGES"
+conda create -n testenv --yes -c $CONDA_CHANNEL $to_install
 conda activate testenv
 
 if [[ -z "$JOBLIB_TESTS" ]]; then

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -708,7 +708,7 @@ class ExecutorTest:
             if bool(sysconfig.get_config_var("Py_GIL_DISABLED")):
                 # XXX: full GC collect is needed on free-threading CPython to
                 # actually trigger the garbage collection of my_object once the
-                # executor is done with it. Note sure if this can be considered
+                # executor is done with it. Not sure if this can be considered
                 # a bug or not.
                 gc.collect()
             collected = collect.wait(timeout=1.0)

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -705,7 +705,7 @@ class ExecutorTest:
 
         collected = False
         for _ in range(5):
-            if bool(sysconfig.get_config_var("Py_GIL_DISABLED")):
+            if hasattr(sys, "_is_gil_enabled") and not sys._is_gil_enabled():
                 # XXX: full GC collect is needed on free-threading CPython to
                 # actually trigger the garbage collection of my_object once the
                 # executor is done with it. Not sure if this can be considered

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -1,3 +1,4 @@
+import sysconfig
 from loky import process_executor
 
 import os
@@ -704,6 +705,12 @@ class ExecutorTest:
 
         collected = False
         for _ in range(5):
+            if bool(sysconfig.get_config_var("Py_GIL_DISABLED")):
+                # XXX: full GC collect is needed on free-threading CPython to
+                # actually trigger the garbage collection of my_object once the
+                # executor is done with it. Note sure if this can be considered
+                # a bug or not.
+                gc.collect()
             collected = collect.wait(timeout=1.0)
             if collected:
                 return

